MantidPlot/src/ProjectRecovery.cpp

```diff
@@ -145,9 +145,14 @@ std::vector<int> orderProcessIDs(std::vector<Poco::Path> paths) {
               // Last modified is first!
               return a1.getLastModified() > b1.getLastModified();
             });
-
   for (auto c : paths) {
-    returnValues.emplace_back(std::stoi(c.directory(c.depth() - 1)));
+    try {
+      returnValues.emplace_back(std::stoi(c.directory(c.depth() - 1)));
+    } catch (std::invalid_argument) {
+      // The folder or file here is not a number (So shouldn't exist) so delete
+      // it recursively
+      Poco::File(c).remove(true);
```

Comment:
Recursively removing files/directories sounds dangerous. Are we absolutely certain that `paths` never contains anything that may cause this to fail and accidentally remove a home directory, recursively? Can we do some sort of sanity check here?